### DISCUSSION TITLE
Migrating from App of Apps approach to ApplicationSet approach

### DISCUSSION THREAD

**Discussion by dmarquez-splunk** [argoproj/argo-cd#19409] [General]
Hi,

Currently we are managing our User's applications via an app-of-apps approach where users create an a config entry in one of our repos, our jsonnet code renders their entry into an application manifest which lives as part of our parent app in argocd, and their resources are then subsequently deployed once their changes are merged and the app is created in the cluster.

I am interested in exploring the process in migrating from this approach to ApplicationSet approach to avoid any downtime or outages of our customer apps? I'd imaging we'd have to do something as follows. My biggest fear is that when we migrate their Application manifest to an ApplicationSet manifest then our Parent app will delete the application from the cluster (which will subsequently delete their resources), apply the ApplicationSet manifest to the cluster, and then ApplicationSet controller will recreate their app out of the ApplicationSet which should then recreate their resources.

Is there anyway to prevent this deletion and recreation of resources? If we leave their app in the cluster and create the ApplicationSet will there be any reconciliation errors we will run into? I'm interested in knowing if anyone as gone through this process and can shed some light on some of these questions. 😄

**Comment by jhanbo** [comment]
You can delete the application without deleting its resources 

`argocd app delete APPNAME --cascade=false`

https://argo-cd.readthedocs.io/en/stable/user-guide/app_deletion/
